src/lib/request-with-rate-limit.ts

```diff
@@ -0,0 +1,37 @@
+import * as needle from 'needle';
+import Bottleneck from 'bottleneck';
+import { OutgoingHttpHeaders } from 'http2';
+
+export async function requestWithRateLimitRetries(
+    verb: needle.NeedleHttpVerbs,
+    url: string,
+    headers: OutgoingHttpHeaders,
+    limiter?: Bottleneck,
+  ): Promise<any> {
+    let data;
+    const maxRetries = 7;
+    let attempt = 0;
+    while (attempt < maxRetries) {
+        limiter ? limiter : limiter = new Bottleneck({ maxConcurrent: 1, minTime: 1000, });
```

Comment:
what about the on.failed etc? https://github.com/snyk-tech-services/snyk-api-import/pull/279/files#diff-fbabb62ba4d3899fc177d504b897abdce9e78e1fa4487eb9f811c6855b669851R19-R27

What happens if limiter fails?

Since this now a wrapper around limiter I would rename the function to `limiterWithRateLimitRetries` and always require the limiter as mandatory so you have to create it before you do many calls not just make it on a one off call.

Reply:
Done

Reply:
limiter.on.failed comes in with the limiter object send by the originating function since we use limiter to pace the calls and each SCM has it's own rate limiting and each script will send a different schedule for limiter to this function.